### ISSUE TITLE
Getting "panic: runtime error" when using minikube service in 1.25.2

### ISSUE THREAD

**Issue by purejgleason** [kubernetes/minikube#13736]
### What Happened?

I create a new minikube cluster

    minikube start --memory=8192 --cpus=4 -n 2

I create a namespace and deploytment/service...

    kubectl create ns foo
    kubectl apply -f test-app/app.yml
    
  

```
app.yml
apiVersion: v1
kind: ServiceAccount
metadata:
  namespace: foo
  name: test-app
---
apiVersion: v1
kind: Service
metadata:
  name: test-app
  namespace: foo
  labels:
    app: test-app
    service: test-app
spec:
  ports:
  - name: http
    port: 8000
    targetPort: 80
  selector:
    app: test-app
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: test-app
  namespace: foo
spec:
  replicas: 1
  selector:
    matchLabels:
      app: test-app
      version: v1
  template:
    metadata:
      labels:
        app: test-app
        version: v1
    spec:
      serviceAccountName: test-app
      containers:
      - image: docker.io/kennethreitz/httpbin
        imagePullPolicy: Always
        name: test-app
        ports:
        - containerPort: 80

```

Next I start up as a service...

    minikube service test-app -n foo --url
    
in `1.25.1` everything runs fine when I try from `1.25.2` I get...

```
minikube service test-app -n foo --url    
😿  service foo/test-app has no node port
panic: runtime error: index out of range [3] with length 3

goroutine 1 [running]:
k8s.io/minikube/cmd/minikube/cmd.glob..func35(0x70d4760?, {0xc000ec42c0?, 0x1, 0x4})
	/Users/.../Code/lib/minikube/cmd/minikube/cmd/service.go:138 +0x645
github.com/spf13/cobra.(*Command).execute(0x70d4760, {0xc000ec4280, 0x4, 0x4})
	/Users/.../go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x70d4260)
	/Users/.../go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x3
```

### Attach the log file

[log.txt](https://github.com/kubernetes/minikube/files/8164510/log.txt)

### Operating System

macOS (Default)

### Driver

Docker

**Comment by purejgleason** [comment]
Also tried compiling from master (and tag) and get the same thing.

**Comment by medyagh** [comment]
sorry for this inconvenience, I am surprised our integration tests didn't catch this. does this happen only when --url is specified?

**Comment by medyagh** [comment]
maybe could be side effect of this PR 
https://github.com/kubernetes/minikube/pull/13367

@ckannon mind taking a look ?

**Comment by medyagh** [comment]
here is the link to the binary from a PR that I think might fix this issue, do you mind trying it out ?

https://storage.googleapis.com/minikube-builds/13756/minikube-linux-amd64
https://storage.googleapis.com/minikube-builds/13756/minikube-darwin-amd64
https://storage.googleapis.com/minikube-builds/13756/minikube-windows-amd64.exe

**Comment by purejgleason** [comment]
Sorry just saw this what would you like me to do @medyagh ?

**Comment by purejgleason** [comment]
NM I get it I will try this out tomorrow sorry it took me so long to respond

**Comment by purejgleason** [comment]
Is it available as a branch?

**Comment by purejgleason** [comment]
LGTM but I would like to compile from source. I'll try that tomorrow.

**Comment by purejgleason** [comment]
Built from source and everything looks to be working thank you.

**Comment by dgoldsmith** [comment]
The updated darwin binary referenced above works for me on Mac OS X 12.2.1 as well.  I was running in to this same issue.  I must have had an older version of minikube initially installed and after removing all Docker/Kubernetes bits from laptop to try again for a clean demo, I got minikube 1.25.2 and ran into the broken service tunnel.  This patch makes it work again.

**Comment by joaquinjsb** [comment]
I'm having the same issue, are u guys gonna publish it as a beta or release it asap ?

thanks

**Comment by dhelfgen** [comment]
I'm having the same issue with windows. How do I use the exe-file? Just replace the exe-file in the installation?

**Comment by Ed87** [comment]
> here is the link to the binary from a PR that I think might fix this issue, do you mind trying it out ?
> 
> https://storage.googleapis.com/minikube-builds/13756/minikube-linux-amd64 https://storage.googleapis.com/minikube-builds/13756/minikube-darwin-amd64 https://storage.googleapis.com/minikube-builds/13756/minikube-windows-amd64.exe

hi @medyagh @all How do I use the exe . I have same issue on Windows 11 minikube install. I have downloaded but it seems its not a click to run type of file. Suggestion is to _replace the existing file_ but I cannot find a similar file in the minikube install path.

Install Path : C:\Users\etshuma_acc\.minikube  (Installed as admin user)

The only reference I got that is similar to the file is :

<img width="626" alt="image" src="https://user-images.githubusercontent.com/23119187/174241813-cf2bce1d-5940-4f4b-a10d-1190fe65dbc6.png">
